src/index.js

```diff
@@ -9,21 +9,23 @@ function yieldTypes(schema) {
   return schema.types;
 }
 
-function filterTypes(whitelist) {
+function filterTypes(whitelistConfig) {
   return function(types) {
-    if (!whitelist) {
+    if (!whitelistConfig) {
       return types;
     }
 
+    const whitelistTypes = Object.getOwnPropertyNames(whitelistConfig);
```

Comment:
I learned a thing! `getOwnPropertyNames` ensures that you're just getting the properties you define on a thing. If something changes the prototype of Object and adds props, this makes sure they aren't captured here. It's only properties declared directly on the instance. It's a subtle difference, but something that seems safer to do in a library?

thoughts?